microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc

```diff
@@ -0,0 +1,153 @@
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12609[WFLY-12609]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1346[EAP7-1346]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+[ ] Engineering
+
+[ x ] QE
+
+=== Affected Projects or Components
+
+The primary integration will be happening by the addition of a new subsystem `microprofile-jwt-smallrye` within https://github.com/wildfly/wildfly[WildFly].
+
+Additional integration will be added within https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron] to bridge into the application server's security framework.  This additional integration is specifically to achieve the delivery of support for MicroProfile JWT 1.1, potential new functionality that we identify would be picked up in future enhancements.
+
+=== Other Interested Projects
+
+The integration will also be making use of https://github.com/smallrye/smallrye-jwt[SmallRye JWT] as well as pulling in the https://github.com/eclipse/microprofile-jwt-auth[Eclipse MicroProfile JWT] API library.
+
+== Requirements
+
+=== Hard Requirements
+
+This enhancement is focused on adding support for the MicroProfile JWT 1.1 specification within WildFly.  During development additional opportunities may be identified to provide additional functionality, any such opportunities will be tracked within the WFLY Jira project and reviewed at the end of this enhancement rather than allowing the scope of this enhancement to grow.
+
+Primarily this enhancement will be delivered using a new subsystem `microprofile-jwt-smallrye` which will be added to WildFly.  The subsystem will install a DeploymentUnitProcessor which will detect if MicroProfile JWT is required and activate a suitable security configuration as described within this proposal.
+
+The new subsystem initially will contain no configuration, it's presence alone will enable support for MicorProfile JWT.
+
+[source,xml]
+----
+<subsystem xmlns="urn:wildfly:microprofile-jwt-smallrye:1.0"/>
+----
+
+Additionally the initial version of this subsystem will expose no operations and no attributes.
+
+The default location to specify the authentication requirements is within the `web.xml` deployment descriptor, the MicorProfile JWT specification add a new annotation `@org.eclipse.microprofile.auth.LoginConfig` which allows a JAX-RS `javax.ws.rs.core.Application` to be annotated to specify the desired `auth-method` and `realm-name`.
+
+During deployment, deployments will be scanned for a class extending `Application` annotated with the new `@LoginConfig` annotation - if the authentication configuration was not previously specified within the `web.xml` the configuration from this annotation will be used instead.  If the authentication configuration has already been specified in the `web.xml` annotation scanning will be skipped as the configuration in the `web.xml` takes precendence over the annotation.
```

Comment:
What configuration has to be set in `web.xml` to enable JWT auth?

Reply:
If the auth-method is set to "MP_JWT" MicroProfile JWT will be activated for the deployment - no other configuration is required.

Reply:
Please include this into the document in next update. This may be important for documentation in the future.

Reply:
@honza-kasik This is what it says on line 68 

"After the previous steps have completed if the resulting `auth-method` for the web application being deployed is `MP-JWT` the MicroProfile JWT integration will be activated"

Reply:
Thank you - it wasn't clear to me that this relates to `web.xml` too. I had just the `@LoginConfig` annotation on my mind and when it wasn't clear to me, other may come and have same experience.

Reply:
Yeah that is why I have tried to redefine it as a sequence of events with activation happening at the end of that sequence.